contracts/CompoundFCM.sol

```diff
@@ -0,0 +1,337 @@
+    trader.updateMarginInScaledYieldBearingTokens(updatedTraderMargin);
+
+    // update trader fixed and variable token balances
+    trader.updateBalancesViaDeltas(fixedTokenDelta, variableTokenDelta);
+
+    // deposit notional executed in terms of cTokens (e.g. cDAI) to fully collateralise the position
+    // we need a number of tokens equal to the variable token delta divided by the exchange rate
+    IERC20Minimal(address(_ctoken)).safeTransferFrom(msg.sender, address(this), yieldBearingTokenDelta);
+
+    // transfer fees to the margin engine (in terms of the underlyingToken e.g. cDAI)
+    underlyingToken.safeTransferFrom(msg.sender, address(_marginEngine), cumulativeFeeIncurred);
+
+    emit FullyCollateralisedSwap(msg.sender, trader.marginInScaledYieldBearingTokens, trader.fixedTokenBalance, trader.variableTokenBalance);
+  }
+
+  /// @notice Get Trader Margin In Yield Bearing Tokens
+  /// @dev this function takes the scaledBalance associated with a trader and multiplies it by the current Exchange Rate to get the balance (margin) in terms of the underlying token
+  /// @param traderMarginInScaledYieldBearingTokens traderMarginInScaledYieldBearingTokens
+  function getTraderMarginInYieldBearingTokens(uint256 traderMarginInScaledYieldBearingTokens) internal view returns (uint256 marginInYieldBearingTokens) {
+    // uint256 currentExchangeRate = _ctoken.exchangeRateStored();
+    // marginInYieldBearingTokens = traderMarginInScaledYieldBearingTokens.rayMul(currentExchangeRate);
+    return traderMarginInScaledYieldBearingTokens; // NO scaling to do here. Delete this function?
+  }
+
+  function getTraderMarginInCTokens(address traderAddress)
+        external
+        view
+        returns (uint256 marginInYieldBearingTokens)
+    {
+        TraderWithYieldBearingAssets.Info storage trader = traders[
+            traderAddress
+        ];
+        marginInYieldBearingTokens = trader.marginInScaledYieldBearingTokens;
+    }
+
+
+  /// @notice Unwind Fully Collateralised Fixed Taker Swap
+  /// @param notionalToUnwind The amount of notional to unwind (stop securing with a fixed rate)
+  /// @param sqrtPriceLimitX96 The sqrt price limit (binary fixed point notation) beyond which the unwind cannot progress
+  function unwindFullyCollateralisedFixedTakerSwap(uint256 notionalToUnwind, uint160 sqrtPriceLimitX96) external override {
+
+    // add require statement and isApproval
+
+    TraderWithYieldBearingAssets.Info storage trader = traders[msg.sender];
+
+    require(trader.variableTokenBalance <= 0, "Trader VT balance positive");
+
+    /// @dev it is impossible to unwind more variable token exposure than the user already has
+    /// @dev hencel, the notionalToUnwind needs to be <= absolute value of the variable token balance of the trader
+    /// @audit-casting variableTokenDelta is expected to be negative here, but what if goes above 0 due to rounding imprecision?
```

Comment:
what are the next steps to resolve the audit casting highlighted issue?

Reply:
We have the same comment in the AaveFCM. My take is that we already `require` that `variableTokenDelta <=0` so the comment can probably just come out of both files, but would value your thoughts on whether the existing `require` is safe to have.

Reply:
comment can come out as said above